truffleruby/src/main/java/org/truffleruby/platform/posix/JNRTrufflePosix.java

```diff
@@ -446,28 +410,77 @@ public int open(CharSequence path, int flags, int perm) {
     @TruffleBoundary
     @Override
     public int write(int fd, byte[] buf, int n) {
+        if ((fd == 1 || fd == 2) && context.getOptions().POLYGLOT_STDIO) {
```

Comment:
That seems a micro optimization which is not worth an exception to standard rules to me.
`context.getOptions().POLYGLOT_STDIO` is just 2 reads, isn't it?